matrix/plugins/glitch/main.py

```diff
@@ -0,0 +1,97 @@
+import asyncio
+import functools
+import logging
+import time
+import yaml
+
+from .selectors import Selectors
+from .actions import Actions
+from .plan import generate_plan, validate_plan
+
+
+log = logging.getLogger("glitch")
+DEFAULT_PLAN_NAME = "glitch_plan.{}.yaml"
+
+
+def default_resolver(model, kind, name):
+    if kind not in ["application", "unit", "model", "controller", "relation"]:
+        return None
+    entities = getattr(model, kind + "s")
```

Comment:
I think maybe a lookup table rather than a str manipulation is better here, but the basic issue remains.